### PR TITLE
TT refactor

### PR DESCRIPTION
**Pull request by jpatton-USGS**
This PR refactors the Travel-Time formats to better fit a RESTful Travel Time Service.

### CODE REVIEW

File: java/src/main/java/gov/usgs/processingformats/TravelTimeRequest.java [hunk 1]
```diff
@@ -12,39 +12,47 @@
...
   /** JSON Keys */
-  public static final String TYPE_KEY = "Type";
+  public static final String SOURCE_KEY = "Source";
 
-  public static final String DISTANCE_KEY = "Distance";
-  public static final String ELEVATION_KEY = "Elevation";
-  public static final String LATITUDE_KEY = "Latitude";
-  public static final String LONGITUDE_KEY = "Longitude";
-  public static final String DATA_KEY = "Data";
+  public static final String Receivers_KEY = "Receivers";
```
Comment (mguy-usgs) [inline java/src/main/java/gov/usgs/processingformats/TravelTimeRequest.java:17]: totally minor, for consistency Receivers_KEY should be all caps RECEIVERS_KEY

File: java/src/main/java/gov/usgs/processingformats/TravelTimeRequest.java [hunk 2]
```diff
@@ -86,49 +102,96 @@ public TravelTimeRequest(
...
-    // Distance
-    if (newJSONObject.containsKey(DISTANCE_KEY)) {
-      Distance = (double) newJSONObject.get(DISTANCE_KEY);
+    // Receivers
+    if (newJSONObject.containsKey(Receivers_KEY)) {
+
+      Receivers = new ArrayList<TravelTimeReceiver>();
+
+      // get the array
+      JSONArray ReceiversArray = (JSONArray) newJSONObject.get(Receivers_KEY);
```
Comment (mguy-usgs) [inline java/src/main/java/gov/usgs/processingformats/TravelTimeRequest.java:118]: change Receivers_KEY to RECEIVERS_KEY everywhere

File: java/src/main/java/gov/usgs/processingformats/TravelTimeRequest.java [hunk 3]
```diff
@@ -228,71 +277,85 @@ public JSONObject toJSON() {
...
 
-    // optional values
-    // Latitude
-    if (Latitude != null) {
-      newJSONObject.put(LATITUDE_KEY, Latitude);
+        ReceiversArray.add(ReceiverObject);
+      }
+
+      if (!ReceiversArray.isEmpty()) {
+        newJSONObject.put(Receivers_KEY, ReceiversArray);
```
Comment (mguy-usgs) [inline java/src/main/java/gov/usgs/processingformats/TravelTimeRequest.java:300]: here too

**Review by mguy-usgs — COMMENTED** [review]
Let me know if you want to talk me through any of your changes, I think we did mostly from the TT service side, and i asked a couple questions that might want changes, so just confirm and let me know and then I'll merge.

File: format-docs/TravelTimePlotRequest.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+
+## Glossary
+
+**Required Input Values:**
+
+These are the values **required** to define a Travel-Time request.
+
+* Source -  A [TravelTimeSource](TravelTimeSource.md) object containing the geographic source information
```
[the remark this answers is not in the document]
Comment (jpatton-USGS) [reply format-docs/TravelTimePlotRequest.md:58]: Yes, but Ray's travel time nomenclature is that phases travel from the source (hypo) to the receiver (station) so I followed that here.

File: format-docs/TravelTimePlotRequest.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+* Receivers - An array of [TravelTimeReceiver](TravelTimeReceiver.md) objects containing the geographic distance (or receiver information) to request travel time information for.
```
[the remark this answers is not in the document]
Comment (jpatton-USGS) [reply format-docs/TravelTimePlotRequest.md:59]: Yes, but Ray's travel time nomenclature is that phases travel from the source (hypo) to the receiver (station) so I followed that here.

File: format-docs/TravelTimeRequest.md [hunk 6]
```diff
@@ -14,92 +14,84 @@ information service
...
 
 **Optional Input Values:**
 
-* Latitude - A number containing the geographic receiver latitude in
-degrees.
-* Longitude - A number containing the geographic receiver longitude in
-degrees.
+* EarthModel - A string containing the earth model to use, defaults to the
+AK135 earth model.
+* PhaseTypes - An array of strings listing the phase types desired. Defaults to all phases.
```
[the remark this answers is not in the document]
Comment (jpatton-USGS) [reply format-docs/TravelTimeRequest.md:84]: Yes, and no, for some phases, if the phase type is not present, then that phase type is not returned, however some phases, like P and S are "generic", and all phases of that type (for P, Pg, Pb, Pn, P, Pdif, PKP*) are returned. I think that sort of info is too involved for the format definitions, but need to be added as documentation in the tt package itself

File: java/src/main/java/gov/usgs/processingformats/TravelTimePlotDataSample.java [hunk 7]
```diff
@@ -29,10 +30,13 @@
   /** Optional statistical Observability of the seismic phase. */
   public Double Observability;
 
+  /** Optional ray parameter in seconds/degree. */
```
[the remark this answers is not in the document]
Comment (jpatton-USGS) [reply java/src/main/java/gov/usgs/processingformats/TravelTimePlotDataSample.java:33]: When generating a travel time, the package shoots utilizes "ray tracing" through the complex geologic structure. The ray parameter is more or less a representation the change of the ray takeoff angle as the ray passes through the geology, gradually becoming horizontal as depth increases. Again, this is more fundamental tt package stuff than format stuff

**Pull request merged**